### PR TITLE
Add iOS Bark notify hook

### PR DESCRIPTION
**Pull request by hguandl**
Bark is an iOS app which takes advantages of APNs (Apple Push Notification service) to send custom notifications conveniently.

bark

This pull request implements the notify hook of Bark API, including the support of sound and group options.

Possible Wiki:
## 16. Set notification for iOS Bark

Send notification via Bark's API. The notification will be pushed to the specified pushover application.

Download Bark from [App Store](https://apps.apple.com/cn/app/bark-%E7%BB%99%E4%BD%A0%E7%9A%84%E6%89%8B%E6%9C%BA%E5%8F%91%E6%8E%A8%E9%80%81/id1403753865).

Open the app and get the API URL. It usually starts with "https://api.day.app/" and followed by a UUID of length 22. Set BARK_API_URL to the complete URL, which is in the format of "https://api.day.app/XXXXXXXXXXXXXXXXXXXXXX".

You can also use your own API server. Please refer to <https://github.com/Finb/bark-server>.

BARK_SOUND is an optional variable to set the sound of notification. Please refer to the app to get all available sounds.

BARK_GROUP is the group name of the notifications, which can be shown in the notification center of iOS. 

```bash
export BARK_API_URL="https://api.day.app/XXXXXXXXXXXXXXXXXXXXXX"
export BARK_SOUND="newmail"
export BARK_GROUP=ACME

acme.sh --set-notify --notify-hook bark
```

### CODE REVIEW

**Comment by Neilpang** [comment]
请把用法直接添加到这个页面末尾:
https://github.com/acmesh-official/acme.sh/wiki/notify

**Comment by hguandl** [comment]
已添加。

File: notify/bark.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# subject  content statusCode
+bark_send() {
+  _subject="$1"
+  _content="$2"
+  _statusCode="$3" #0: success, 1: error 2($RENEW_SKIP): skipped
+  _debug "_subject" "$_subject"
+  _debug "_content" "$_content"
+  _debug "_statusCode" "$_statusCode"
+
+  BARK_API_URL="${BARK_API_URL:-$(_readaccountconf_mutable BARK_API_URL)}"
```
Comment (Neilpang) [inline notify/bark.sh:18]: 如果没有指定 BARK_API_URL  是不是可以使用默认值 api.day.app

这个默认服务器可以用吗?

Comment (hguandl) [reply notify/bark.sh:18]: 默认服务器可以使用，但是后面的UUID是每个设备独立的，需要指定。完整的 URL 是 API 服务器（默认 api.day.app）和 UUID 组合。也可以考虑把 UUID 提取出来当单独的配置项，然后 API 服务器设置为默认。但是 app 自己提供了一个一件复制完整 URL 的按钮，考虑到使用的便利性就干脆用完整 URL 了。

IMG_0438

**Pull request merged**